docs/docs/examples/full-pipelines/dspy/index.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Solving NYT Connections with DsPy
+description: Learn how to build an AI puzzle solver using DsPy and Dagster
+last_update:
+  author: Dennis Hume
+sidebar_custom_props:
+  logo: images/integrations/dspy.png
+tags: [code-example]
+canonicalUrl: '/examples/dspy'
+slug: '/examples/dspy'
+---
+
+# Solving NYT Connections with DsPy
+
+In this tutorial, you'll build an AI system that solves [NYT Connections puzzles](https://www.nytimes.com/games/connections) using [DsPy](https://dspy.ai/) for structured reasoning and Dagster for pipeline orchestration. DsPy is a framework for building modular AI software that can be compiled into effective prompts and optimized automatically.
+
+This example demonstrates how to:
+
+- Load and process Connections puzzle data
+- Build a DsPy solver module with game logic
+- Optimize the solver using MIPROv2 automatic optimization
+- Evaluate puzzle-solving performance with custom metrics
+- Deploy and monitor AI systems in production
+
+You will learn to:
```

Comment:
This intro runs a little long -- I think you could cut the "You will learn to" section without loss of clarity for the reader.